cli/ops/os.rs

```diff
@@ -57,17 +57,93 @@ fn op_start(
   })))
 }
 
-fn op_home_dir(
+#[derive(Deserialize)]
+struct GetDirArgs {
+  name: std::string::String,
+}
+
+fn op_get_dir(
   state: &ThreadSafeState,
-  _args: Value,
+  args: Value,
   _zero_copy: Option<PinnedBuf>,
 ) -> Result<JsonOp, ErrBox> {
   state.check_env()?;
-  let path = dirs::home_dir()
-    .unwrap_or_default()
-    .into_os_string()
-    .into_string()
-    .unwrap_or_default();
+  let args: GetDirArgs = serde_json::from_value(args)?;
+
+  let path = match args.name.as_str() {
+    "home" => dirs::home_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "config" => dirs::config_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "cache" => dirs::cache_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "data" => dirs::data_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "data_local" => dirs::data_local_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "audio" => dirs::audio_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "desktop" => dirs::desktop_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "document" => dirs::document_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "download" => dirs::download_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "font" => dirs::font_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "picture" => dirs::picture_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "public" => dirs::public_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "template" => dirs::template_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
+    "video" => dirs::video_dir()
+      .unwrap_or_default()
+      .into_os_string()
+      .into_string()
+      .unwrap_or_default(),
```

Comment:
It seems you could reduce code here by lifting the unwrap_or_default().into_os_string()... into the next statement.